File: Data Structures & Algorithms/max-consecutive-ones/submission-0.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMaxConsecutiveOnes(nums: number[]): number {
        let bufferCount = 0;
        let max = 0
        const targetNum = 1;

        for (let num = 0; num <= nums.length; num++) {
            if (nums[num] === targetNum) {
                bufferCount += 1
            } else {
                max = bufferCount > max ? bufferCount : max;
                bufferCount = 0
            }
        }   


        return max
    }
}

File: Data Structures & Algorithms/largest-unique-number/submission-0.py
class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        nums_map = {}
        res = -1

        for num in nums:
            if num in nums_map:
                nums_map[num] += 1
            else:
                nums_map[num] = 1
        
        for num in nums_map:
            if nums_map[num] == 1:
                res = max(res, num)

        return res